### ISSUE TITLE
E402 false positive with cell magics in Jupyter notebooks

### ISSUE THREAD

**Issue by jamesmyatt** [astral-sh/ruff#10357]
The E402 rule should permit cell magics as well as comments before the import statements. The cell magics must come first.

e.g.

```
%%time
import expensive_module
```

I suspect that line magics should also be permitted before import statements, like:

```
%time import expensive_module
import cheap_module
```

Error message is:

```
E402 Module level import not at top of cell
```

**Comment by dhruvmanila** [comment]
Yeah, I agree. Certain cell magics like the one you mentioned should come before the actual code.

I don't think we flag `E402` if there's a comment before the first import: https://play.ruff.rs/8d6055cc-be7c-41d9-aac3-ef991213116c
